### PR TITLE
Enhances XDSL readwrite methods and tests

### PR DESCRIPTION
**Pull request by Nimish-4**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.

### CODE REVIEW

File: pgmpy/readwrite/XDSL.py [hunk 1]
```diff
@@ -361,7 +360,7 @@ def get_cpds(self):
 
             # Add the <probabilities> element.
             probs_elem = etree.SubElement(cpt_elem, "probabilities")
-            values = np.array(cpd.get_values())
+            values = compat_fns.to_numpy(cpd.get_values())
```
Comment (ankurankan) [inline pgmpy/readwrite/XDSL.py]: @Nimish-4 Shouldn't this be dependent on the compute backend? A numpy array if the backend is numpy otherwise a torch Tensor.

File: pgmpy/utils/compat_fns.py [hunk 2]
```diff
@@ -21,7 +21,12 @@ def copy(arr):
         elif isinstance(arr, (int, float)):
             return deepcopy(arr)
     else:
-        return torch.tensor(arr, dtype=config.get_dtype(), device=config.get_device())
+        if isinstance(arr, torch.Tensor):
+            return arr.detach().clone()
```
Comment (ankurankan) [inline pgmpy/utils/compat_fns.py:25]: I am not sure why this is needed. Is there a case where the previous version breaks? If the problem is that the type and device gets changed, I think a better solution would be to just replace the previous code with `detach().clone()`.

**Review by ankurankan — CHANGES_REQUESTED** [review]

File: pgmpy/readwrite/XDSL.py (hunk 1 above)
Comment (Nimish-4) [reply pgmpy/readwrite/XDSL.py]: @ankurankan Yeah, the `cpd.get_values()` should return either a numpy or a Torch array depending on the backend. I just thought there would not be much use of a Torch array here and kept the original behaviour from the code. I'll make the change for consistency.

File: pgmpy/utils/compat_fns.py (hunk 2 above)
Comment (Nimish-4) [reply pgmpy/utils/compat_fns.py:25]: In the normal `torch.tensor()` case, it threw a user warning when `arr` is a torch tensor itself. So I replaced it with `.detach().clone()` . The second case is to handle when arr is some dtype other than a torch tensor ... I think such a situation did come up in of the tests somewhere.

**Review by ankurankan — APPROVED** [review]
